src/main/java/teammates/ui/newcontroller/FeedbackQuestionInfo.java

```diff
@@ -0,0 +1,421 @@
+        public void validate() {
+            assertTrue(questionNumber >= 1, "Invalid question number");
+            assertTrue(questionBrief != null, "Question brief cannot be null");
+            assertTrue(!questionBrief.isEmpty(), "Question brief cannot be empty");
+            assertTrue(questionDetails != null, "Question details cannot be null");
+
+            assertTrue(questionType != null, "Question type cannot be null");
+            assertTrue(giverType != null, "Giver type cannot be null");
+            assertTrue(recipientType != null, "Recipient type cannot be null");
+
+            assertTrue(numberOfEntitiesToGiveFeedbackToSetting != null,
+                    "numberOfEntitiesToGiveFeedbackToSetting cannot be null");
+            if (numberOfEntitiesToGiveFeedbackToSetting == NumberOfEntitiesToGiveFeedbackToSetting.CUSTOM) {
+                assertTrue(customNumberOfEntitiesToGiveFeedbackTo != null,
+                        "customNumberOfEntitiesToGiveFeedbackTo must be set");
+            }
+
+            assertTrue(showResponsesTo != null, "showResponsesTo cannot be null");
+            assertTrue(showGiverNameTo != null, "showGiverNameTo cannot be null");
+            assertTrue(showRecipientNameTo != null, "showRecipientNameTo cannot be null");
+        }
+
+        public int getQuestionNumber() {
+            return questionNumber;
+        }
+
+        public String getQuestionDescription() {
+            return questionDescription;
+        }
+
+        public FeedbackQuestionDetails getQuestionDetails() {
+            FeedbackQuestionDetails details =
+                    JsonUtils.fromJson(JsonUtils.toJson(questionDetails), questionType.getQuestionDetailsClass());
+            details.setQuestionText(questionBrief);
+            return details;
+        }
+
+        public FeedbackQuestionType getQuestionType() {
+            return questionType;
+        }
+
+        public FeedbackParticipantType getGiverType() {
+            return giverType;
+        }
+
+        public FeedbackParticipantType getRecipientType() {
+            return recipientType;
+        }
+
+        public int getNumberOfEntitiesToGiveFeedbackTo() {
+            switch (numberOfEntitiesToGiveFeedbackToSetting) {
+            case CUSTOM:
+                return customNumberOfEntitiesToGiveFeedbackTo;
+            case UNLIMITED:
+                return Const.MAX_POSSIBLE_RECIPIENTS;
+            default:
+                Assumption.fail("Unknown numberOfEntitiesToGiveFeedbackToSetting"
+                        + numberOfEntitiesToGiveFeedbackToSetting);
+                break;
+            }
+            return 0;
+        }
+
+        public List<FeedbackParticipantType> getShowResponsesTo() {
+            List<FeedbackParticipantType> showResponsesTo =
+                    this.convertToFeedbackParticipantType(this.showResponsesTo);
+
+            // specially handling for contribution questions
+            // TODO: remove the hack
```

Comment:
The hack exists in V6 frontend. I don't want to introduce the complexity in V7 frontend so I put it to the backend.

It might be addressed after V7 or at later stage of V7 I think. This will be a separate issue.